src/components/RoundForm.vue

```diff
@@ -0,0 +1,264 @@
+            <input v-model.number="score.strokes[num - 1]" number 
+            type="number" style="width: 35px;" min="0" max="20">
+          </td>
+        </tr>
+        <tr v-show="canAddScores">
+          <td>
+            <select v-model="newPlayer">
+              <option value="" default selected disabled>Select player</option>
+              <option v-for="player in newPlayerOptions" :key="player">
+                {{ player }}
+              </option>
+            </select>
+            <button @click="addScore" :disabled="newPlayer == ''">Add</button>
+          </td>
+        </tr>
+      </table><br><br>
+    </div>
+
+    <!-- wrap the button in a div so that tooltip can be showed
+    when button is disabled -->
+    <div v-tooltip="submitWarningMsg">
+      <button v-on:click="submit"
+      :disabled="submitWarningMsg != ''">
+        Submit
+      </button>
+    </div>
+
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'RoundForm',
+  data: function() {
+    return {
+      msg: 'Getting input data ...',
+      date: '',
+      club: {},
+      courses: [],
+      newCourse: {},
+      newPlayer: '',
+      clubOptions: [],
+      courseOptions: [],
+      playerOptions: [],
+    }
+  },
+  computed: {
+    newCourseOptions () {
+      let existingCourses = this.courses.map(c => c.name)
+      return this.courseOptions.filter(c => !existingCourses.includes(c.name))
+    },
+    canAddCourses () {
+      return this.courses.length < 2
+    },
+    newPlayerOptions () {
+      if (this.courses.length == 0) {
+        return []
+      }
+      let players = this.courses[0].scores.map(s => s.player)
+      return this.playerOptions.filter(p => !players.includes(p))
+    },
+    canAddScores () {
+      return this.courses[0].scores.length < 4
+    },
+    submitWarningMsg () {
+      if (!this.date) return 'Missing date'
+      if (Object.keys(this.club).length == 0) return 'Missing club'
+      if (this.courses.length == 0)  return 'Missing course'
+      if (!this.courses.every(c => c.tees)) return 'Missing tees'
+      if (!this.courses.every(c => c.numHoles > 0)) return 'Missing holes'
+      if (!this.courses.every(c => c.scores.length > 0)) {
+        return 'Missing players'
+      }
+      let strokes = this.courses.flatMap(c => c.scores.flatMap(s => s.strokes))
+      if (strokes.includes(0)) {
+       return 'Missing scores. All scores must be greater than 0'
+      }
+      return ''
+    },
+  },
+  created () {
+    this.$http.get(
+      '/api/input-rounds'
+    ).then(function (response) {
+      this.clubOptions = response.data.clubs;
+      this.playerOptions = response.data.players;
+      this.msg = 'Input new round below.'
+    }, function (error) {
+      console.log(error.data);
+      alert(error.data);
+    })
+  },
+  methods: {
+    changeClub: function() {
+      this.courses = []
+      this.courseOptions = this.club.courses
+    },
+    addCourse: function() {
+      this.courses.push({
```

Comment:
This is messy creating all these nested fields, could clean it up in the future by adding a `Course` component